ENIGMAsystem/SHELL/Platforms/Win32/WINDOWSfilemanip.cpp

```diff
@@ -162,10 +164,11 @@ int file_copy(std::string fname, std::string newname) {
 }
 
 int directory_exists(std::string dname) {
-  DWORD dwAttrib = GetFileAttributes(dname.c_str());
-
-  return (dwAttrib != INVALID_FILE_ATTRIBUTES &&
-         (dwAttrib & FILE_ATTRIBUTE_DIRECTORY));
+  DWORD file_attr;
+  tstring tstr_dname = widen(dname);
+  file_attr = GetFileAttributesW(tstr_dname.c_str());
+  return (file_attr != INVALID_FILE_ATTRIBUTES &&
```

Comment:
You could have done this for `file_exists` too. §4.7/4 of the C++ standard states the following:
>If the source type is bool, the value **false** is converted to zero and the value **true** is converted to one.